lib/compress/zstd_compress.c

```diff
@@ -2453,6 +2453,16 @@ size_t ZSTD_compressBegin_usingCDict_advanced(
     if (cdict==NULL) return ERROR(dictionary_wrong);
     {   ZSTD_CCtx_params params = cctx->requestedParams;
         params.cParams = ZSTD_getCParamsFromCDict(cdict);
+        /* Increase window log to fit the entire dictionary and source if the
+         * source size is known. Limit the increase to 19, which is the
+         * window log for compression level 1 with the largest source size.
+         */
+        if (pledgedSrcSize != ZSTD_CONTENTSIZE_UNKNOWN) {
+            size_t const totalSize = cdict->dictContentSize + pledgedSrcSize;
```

Comment:
`totalSize` is not necessary.
The full dictionary remains accessible as long as data being processed is `< windowSize`.
So you can base `windowLog` on `pledgedSrcSize` only.

Also : beware of `pledgedSrcSize` overflow, as it is a 64-bits value.